datafusion/src/avro_to_arrow/arrow_array_reader.rs

```diff
@@ -112,6 +112,7 @@ impl<'a, R: Read> AvroArrowArrayReader<'a, R> {
             // reached end of file
             return Ok(None);
         }
+        let rows = rows.iter().collect::<Vec<&Vec<(String, Value)>>>();
```

Comment:
Can we reduce this to one iteration by doing something in the loop above (`self.reader.by_ref().take(batch_size).map(|value|...` )

Reply:
Unfortunately, in both cases we have to take ownership of the values from the iterator first. 
I couldn't find a way to do otherwise...